eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -100,20 +103,152 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
+        "$match":
+            {
+                "tax": {
+                    "$eq": taxonomy,
+                },
+                "asm": {
+                    "$in": assembly_list,
+                }
+            }
+    }
+    group_by_assembly = {
+        "$group":
+            {
+                "_id": {"assembly": "$asm"},
+                "rs_count": {"$sum": 1},
+            }
+    }
+    return [filter_by_taxonomy_and_assembly, group_by_assembly]
+
+
+def get_rs_count_per_assembly_in_collection(mongo_source, coll_name, pipeline):
+    coll = mongo_source.mongo_handle[mongo_source.db_name][coll_name]
+    agg_result = coll.aggregate(pipeline, batchSize=0)
+    agg_res_dict = {}
+    for entry in agg_result:
+        agg_res_dict[entry['_id']['assembly']] = entry['rs_count']
+    return agg_res_dict
+
+
+def get_sources_and_rs_count(cve_res, dbsnp_res, assembly):
+    if assembly in cve_res and assembly in dbsnp_res:
+        return ['DBSNP, EVA', cve_res[assembly] + dbsnp_res[assembly]]
+    elif assembly in cve_res and assembly not in dbsnp_res:
+        return ['EVA', cve_res[assembly]]
+    elif assembly not in cve_res and assembly in dbsnp_res:
+        return ['DBSNP', dbsnp_res[assembly]]
+    else:
+        # if there is an assembly in evapro but there is no clustered variant in any of the collections,
+        # an entry will still be created in clustering tracking table with sources as None and rs count 0
+        return ['None', 0]
+
+
+def check_if_entry_exists_for_taxonomy_and_assembly(pg_conn, taxonomy, assembly):
+    logger.info(f'check if entry exists for taxonomy({taxonomy}) and assembly({assembly})')
+    query_check = f"""SELECT count(*) from eva_progress_tracker.clustering_release_tracker
+                                 WHERE taxonomy={taxonomy} and assembly_accession='{assembly}'"""
+    result = get_all_results_for_query(pg_conn, query_check)
+    return True if result[0][0] != 0 else False
+
+
+def update_rs_count_for_taxonomy_assembly(pg_conn, rs_count, taxonomy, assembly):
+    logger.info(f'updating rs count({rs_count}) for taxonomy({taxonomy}) and assembly({assembly})')
+    query_update = (f"""UPDATE eva_progress_tracker.clustering_release_tracker SET num_rs_to_release={rs_count}
+                        WHERE taxonomy={taxonomy} and assembly_accession='{assembly}'""")
+    execute_query(pg_conn, query_update)
+
+
+def insert_new_entry_for_taxonomy_assembly(pg_conn, sources, rs_count, release_version, taxonomy, assembly, reference_directory):
+    logger.info(f'inserting rs count({rs_count}) for taxonomy({taxonomy}) and assembly({assembly})')
+    scientific_name = get_scientific_name(pg_conn, taxonomy)
+    release_folder_name = normalise_taxon_scientific_name(scientific_name)
+    fasta_path = NCBIAssembly(assembly, scientific_name, reference_directory).assembly_fasta_path
+    report_path = NCBIAssembly(assembly, scientific_name, reference_directory).assembly_report_path
```

Comment:
My fault I should have been more explicit:
```suggestion
    ncbi_assembly = NCBIAssembly(assembly, scientific_name, reference_directory)
    fasta_path = ncbi_assembly.assembly_fasta_path
    report_path = ncbi_assembly.assembly_report_path
```

Reply:
updated